File: app/src/main/java/com/natateam/myevents/adapter/MyRecyclerViewAdapter.java
package com.natateam.myevents.adapter;

import android.support.constraint.ConstraintLayout;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.natateam.myevents.Consts;
import com.natateam.myevents.MainActivity;
import com.natateam.myevents.R;
import com.natateam.myevents.TimeUtils;
import com.natateam.myevents.db.Event;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.realm.OrderedRealmCollection;

/**
 * Created by macbook on 07/07/ 15.
 */
public class MyRecyclerViewAdapter extends RealmRecyclerViewAdapter<Event, MyRecyclerViewAdapter.MyViewHolder> {

private final MainActivity activity;

public MyRecyclerViewAdapter(MainActivity activity, OrderedRealmCollection<Event> data) {
        super(activity ,data, true);
        this.activity = activity;
        }

@Override
public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = inflater.inflate(R.layout.event_item, parent, false);
        return new MyViewHolder(itemView);
        }

@Override
public void onBindViewHolder(MyViewHolder holder, int position) {
        Event obj = getData().get(position);
        boolean isTop = true;
        if (position>0) {
         Event prevEvent = getData().get(position-1);
         long currentBeginOfDay = TimeUtils.getBeginOfDaInMillis(obj.getDateMS());
         long prevBeginOfDay = TimeUtils.getBeginOfDaInMillis(prevEvent.getDateMS());
         if (currentBeginOfDay==prevBeginOfDay){
             isTop=false;
         }
        }
        holder.setItem(obj,isTop);
        }

class MyViewHolder extends RecyclerView.ViewHolder implements View.OnLongClickListener,View.OnClickListener {
    @BindView(R.id.txt_name)
    public TextView title;
    @BindView(R.id.txt_type)
    public TextView txtType;
    @BindView(R.id.txt_description)
    public TextView txtDescription;
    @BindView(R.id.layout_top_date)
    ConstraintLayout layoutTopDate;
    @BindView(R.id.txt_date)
    TextView txtDate;
    @BindView(R.id.txt_date_string)
    TextView txtDateString;
    public Event data;


    public MyViewHolder(View view) {
        super(view);
        ButterKnife.bind(this,view);
        view.setOnLongClickListener(this);
        view.setOnClickListener(this);
    }

    public void setItem(Event obj, boolean isTop){
        data = obj;
        title.setText(obj.getTitle());
        txtDescription.setText(obj.getDesc());
        if  (isTop){
            layoutTopDate.setVisibility(View.VISIBLE);
            txtDate.setText(TimeUtils.getFormatDateFromLong(obj.getDateMS(), Consts.DATE_FORMAT));
            txtDateString.setText(TimeUtils.getFormatDateFromLong(obj.getDateMS(), Consts.DATE_STRING_FORMAT));
        }else {
            layoutTopDate.setVisibility(View.GONE);
        }
    }

    @Override
    public boolean onLongClick(View v) {
        //activity.deleteItem(data);
        return true;
    }


    @Override
    public void onClick(View v) {
        activity.showEventActivityWithId(data.getId());
    }
}
}

File: app/src/main/java/com/natateam/myevents/adapter/MyRecyclerViewAdapter.kt
package com.natateam.myevents.adapter

import android.content.Context
import android.support.constraint.ConstraintLayout
import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.natateam.myevents.Consts
import com.natateam.myevents.R
import com.natateam.myevents.TimeUtils
import com.natateam.myevents.db.Event
import io.realm.OrderedRealmCollection
import org.jetbrains.anko.find

/**
 * Created by macbook on 07/07/ 15.
 */
class MyRecyclerViewAdapter(private val context: Context, data: OrderedRealmCollection<Event>, val onClick: (Event) -> Unit) : RealmRecyclerViewAdapter<Event, MyViewHolder>(context, data, true) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val itemView = inflater.inflate(R.layout.event_item, parent, false)
        return MyViewHolder(itemView, onClick)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val obj = data!![position]
        var isTop = true
        if (position > 0) {
            val prevEvent = data!![position - 1]
            val currentBeginOfDay = TimeUtils.getBeginOfDaInMillis(obj.dateMS)
            val prevBeginOfDay = TimeUtils.getBeginOfDaInMillis(prevEvent.dateMS)
            if (currentBeginOfDay == prevBeginOfDay) {
                isTop = false
            }
        }
        holder.setItem(obj, isTop)
    }


}

class MyViewHolder(val view: View, val onClick: (Event) -> Unit) : RecyclerView.ViewHolder(view) {
    var txtTitle: TextView? = null
    var txtType: TextView? = null
    var txtDescription: TextView? = null
    var layoutTopDate: ConstraintLayout? = null
    var txtDate: TextView? = null
    var txtDateString: TextView? = null


    init {
        txtTitle = view.find(R.id.txt_name)
        txtType = view.find(R.id.txt_type)
        txtDescription = view.find(R.id.txt_birthday)
        layoutTopDate = view.find(R.id.layout_top_date)
        txtDate = view.find(R.id.txt_date)
        txtDateString = view.find(R.id.txt_date_string)
    }

    fun setItem(obj: Event, isTop: Boolean) {
        with(obj) {
            txtTitle!!.text = title
            txtDescription!!.text = desc
            if (isTop) {
                layoutTopDate!!.visibility = View.VISIBLE
                txtDate!!.text = TimeUtils.getFormatDateFromLong(dateMS, Consts.DATE_FORMAT)
                txtDateString!!.text = TimeUtils.getFormatDateFromLong(dateMS, Consts.DATE_STRING_FORMAT)
            } else {
                layoutTopDate!!.visibility = View.GONE
            }
            txtType!!.text = task_time
            view.setOnClickListener { onClick(this) }
        }
    }

}



